frontend/components/MainPage/header/index.tsx

```diff
@@ -38,12 +39,20 @@ export const MainHeader = () => {
   const [isFirstRunModalOpen, setIsFirstRunModalOpen] = useState(false);
   const handleModalClose = useCallback(() => setIsFirstRunModalOpen(false), []);
 
+  const { selectedService } = useServices();
+  const configId = selectedService?.service_config_id;
+  const { isLoaded: isServiceLoaded } = useService({
+    serviceConfigId: configId,
+  });
+  const { isActiveStakingProgramLoaded } = useStakingProgram();
+
   useSetupTrayIcon();
+  // TODO: support loading state
 
   return (
     <Flex justify="start" align="center" gap={10}>
       <AgentHead />
-      <AgentButton />
+      {isServiceLoaded && isActiveStakingProgramLoaded && <AgentButton />}
```

Comment:
the AgentButton checks loading states, at least for service, and renders button with loading state. while loading this now will be empty, is that expected?